wifiphisher/extensions/lure10.py

```diff
@@ -6,101 +6,108 @@
-        :param packet: A scapy.layers.RadioTap object.
+        :param self: A Lure10 object
+        :param packet: A scapy.layers.RadioTap object
         :type self: Lure10
         :type packet: scapy.layers.RadioTap
-        :return: list with the crafted beacon frames
-        :rtype: list
+        :return: A tuple containing ["*"] followed by a list of
+            the crafted beacon frames
+        :rtype: tuple(list, list)
+        .. warning: pkt is not used here but should not be removed since
+            this prototype is requirement
         """
 
-        beacons = []
-
-        if self.first:
-            if self.data.args.lure10_exploit:
-                area_file = constants.LOCS_DIR + self.data.args.lure10_exploit
-                with open(area_file) as a_file:
-                    wlans = [x.strip() for x in a_file.readlines()]
-                    for wlan in wlans:
-                        bssid, essid = wlan.split(' ', 1)
-                        # Frequency for channel 7
-                        frequency = struct.pack("<h", 2407 + 7 * 5)
-                        ap_rates = "\x0c\x12\x18\x24\x30\x48\x60\x6c"
-                        frame = dot11.RadioTap(len=18,
-                                               present='Flags+Rate+Channel+dBm_AntSignal+Antenna',
-                                               notdecoded='\x00\x6c' + \
-                                               frequency + '\xc0\x00\xc0\x01\x00\x00') \
-                                               / dot11.Dot11(subtype=8, \
-                                               addr1='ff:ff:ff:ff:ff:ff', \
-                                               addr2=bssid, \
-                                               addr3=bssid) / dot11.Dot11Beacon(cap=0x2105) \
-                                               / dot11.Dot11Elt(ID='SSID', \
-                                               info="") / dot11.Dot11Elt(ID='Rates', \
-                                               info=ap_rates) / dot11.Dot11Elt(ID='DSset', \
-                                               info=chr(7))
-                        beacons.append(frame)
-            self.beacons_num = len(beacons)
-            self.first = False
-
-        return (["*"], beacons)
+        # only run this code once
+        if self.first_run and self.data.args.lure10_exploit:
+
+            # setup our data structures inside the if for a better performance
+            beacons = list()
+            bssid = str()
+
+            # locate the lure10 file
+            area_file = constants.LOCS_DIR + self.data.args.lure10_exploit
+
+            # open the file
+            with open(area_file) as _file:
+                for line in _file:
+                    # remove any white space and store the bssid(fist word)
+                    line.strip()
+                    bssid = line.split(" ", 1)[0]
+
+                    # craft the required packet parts
+                    frame_part_0 = dot11.RadioTap()
```

Comment:
Instead of writing the complete frame at once, they are separated into multiple parts for both easier reading and editing.